### DISCUSSION TITLE
History variables in a kernel

### DISCUSSION THREAD

**Discussion by etauq** [idaholab/moose#17644] [Q&A General] [answered]
Hi,

I was wondering if we can require history variables in a kernel. If possible, how can we do that? 

Thanks,
Jack

**Comment by YaqiWang** [comment]
If you want your kernel to couple old solutions of variables in a transient calculation, you can add `implicit = false` in the kernel parameters in the input.

**Comment by jiangwen84** [comment, marked as answer]
If you want to access old solution in additional to the current solution, you can use `coupledValueOld(var_name)`
